auth-service/routes/auth.js

```diff
@@ -0,0 +1,46 @@
+const router = require('express').Router();
+const User = require('../models/User');
+const bcrypt = require('bcryptjs');
+const { registerValidation, loginValidation}= require('../validation');
+
+
+router.post('/register',  async(req, res) => {
+    // validate data
+    const{error} = registerValidation(req.body)
+    if(error) return res.status(400).send(error.details[0].message);
+    // check if user exists
+    const emailExist = await User.findOne({email: req.body.email});
```

Comment:
@moyinmi, when you have code that is asynchronous or has a [side effect](https://www.educative.io/answers/what-is-a-side-effect), it's always advisable to wrap the code in a `try-catch` block.